public/app/features/alerting/unified/components/rule-editor/AnnotationsStep.tsx

```diff
@@ -35,31 +35,26 @@ const AnnotationsStep = () => {
   const { fields, append, remove } = useFieldArray({ control, name: 'annotations' });
 
   const selectedDashboardUid = annotations.find((annotation) => annotation.key === Annotation.dashboardUID)?.value;
-  const selectedPanelId = annotations.find((annotation) => annotation.key === Annotation.panelID)?.value;
+  const selectedPanelId = Number(annotations.find((annotation) => annotation.key === Annotation.panelID)?.value);
 
-  const [selectedDashboard, setSelectedDashboard] = useState<DashboardDataDTO | undefined>(undefined);
+  const [selectedDashboard, setSelectedDashboard] = useState<DashboardModel | undefined>(undefined);
   const [selectedPanel, setSelectedPanel] = useState<PanelDTO | undefined>(undefined);
 
-  const { useDashboardQuery } = dashboardApi;
-
-  const { currentData: dashboardResult, isFetching: isDashboardFetching } = useDashboardQuery(
-    { uid: selectedDashboardUid ?? '' },
-    { skip: !selectedDashboardUid }
-  );
+  const { dashboardModel, isFetching: isDashboardFetching } = useDashboardQuery(selectedDashboardUid);
 
   useEffect(() => {
-    if (isDashboardFetching) {
+    if (isDashboardFetching || !dashboardModel) {
       return;
     }
 
-    setSelectedDashboard(dashboardResult?.dashboard);
+    setSelectedDashboard(dashboardModel);
 
-    const allPanels = mergePanels(dashboardResult);
-    const currentPanel = allPanels.find((panel) => panel.id.toString() === selectedPanelId);
+    const allPanels = getVisualPanels(dashboardModel);
+    const currentPanel = allPanels.find((panel) => panel.id === selectedPanelId);
```

Comment:
What happen if selectedPanel does not have id, and there are several panels without id?